biosimulations_runutils/biosim_pipeline/hdf5_compare.py

```diff
@@ -34,28 +34,45 @@ def get_results(results_zip_file: Path) -> dict[str, dict[str, np.ndarray]]:
 
 
 def compare_arrays(arr1: np.ndarray, arr2: np.ndarray) -> bool:
+    # np.seterr(divide='raise')
     if type(arr1[0]) == np.float64:
-        max1 = max(arr1)
-        max2 = max(arr2)
-        atol = max(1e-3, max1*1e-5, max2*1e-5)
-        return np.allclose(arr1, arr2, rtol=1e-4, atol=atol)
+        if np.isnan(arr1).any() or np.isnan(arr2).any():
+            return False, 1e10
+        max1 = np.nanmax(arr1)
+        max2 = np.nanmax(arr2)
+        atol = np.nanmax([1e-3, max1*1e-5, max2*1e-5])
+        rtol = 1e-4
+        try:
+            score = np.nanmax(abs(arr1 - arr2) / (atol + rtol * abs(arr2)))
+        except FloatingPointError as e:
+            print(e)
+        close = np.allclose(arr1, arr2, rtol=rtol, atol=atol, equal_nan=False)
+        assert(score <= 1) == close
+        return close, score
```

Comment:
what happens if you get a FloatingPointError above, it seems that score will not be initialized and you'll get a `UnboundLocalError` if you try to reference the score, maybe set score to a large value when you catch the FloatingPointError.

Reply:
Done!